cmd/nerdctl/save.go

```diff
@@ -105,17 +105,59 @@ func saveImage(images []string, out io.Writer, saveOpts []archive.ExportOpt, cmd
 	saveOpts = append(saveOpts, archive.WithPlatform(platMC))
 
 	imageStore := client.ImageService()
+	var foundImageNames []string
+	var foundID []string
+	walker := &imagewalker.ImageWalker{
+		Client: client,
+		OnFound: func(ctx context.Context, found imagewalker.Found) error {
+			imgName := found.Image.Name
+			imgDigest := found.Image.Target.Digest.String()
+			if imgName != "" && imgDigest != "" {
+				foundImageNames = append(foundImageNames, imgName)
+				foundID = append(foundID, imgDigest)
+				if err = checkRepeatID(foundID); err != nil {
```

Comment:
Why need it?

Reply:
```
[root@k8s nmx]# nerdctl  -n k8s.io images|grep bc
calico/node                           v3.19.1                                                             bc4a631d553b    28 hours ago    linux/amd64    160.6 MiB    56.0 MiB
calico/pod2daemon-flexvol             v3.19.1                                                             bcac7dc4f130    28 hours ago    linux/amd64    20.7 MiB     8.9 MiB
sha256                                5660150975fb82f006d013fdb583b628d3d340ff64775ca3111ca28f523e32df    bcac7dc4f130    28 hours ago    linux/amd64    20.7 MiB     8.9 MiB
sha256                                c4d75af7e098eef5142c65b8fab5b3ff0e5c614ce72e91b8c44ba20249b596ef    bc4a631d553b    28 hours ago    linux/amd64    160.6 MiB    56.0 MiB
```
if i delete it 
nerdctl  -n k8s.io save bca -o test.tar will not show error

Reply:
```go
 108     var foundID map[string]struct{}
 109     walker := &imagewalker.ImageWalker{
 110         Client: client,
 111         OnFound: func(ctx context.Context, found imagewalker.Found) error {
 112             imgName := found.Image.Name
 113             imgDigest := found.Image.Target.Digest.String()
 114             if _, ok := foundID[imgDigest]; !ok && len(foundID) == 1 {
 115                 return fmt.Errorf("ambiguous digest string")
 116             }
 117
 118             foundID[imgDigest] = struct{}{}
 119             saveOpts = append(saveOpts, archive.WithImage(imageStore, imgName))
 120             return nil
 121         },
 122     }
 123     for _, img := range images {
 124         foundID = make(map[string]struct{})
 125         n, err := walker.Walk(ctx, img)
 126         if err != nil {
 127             return err
 128         }
 129         if n == 0 {
 130             return fmt.Errorf("no such image: %s", img)
 131         }
 132     }
```

Reply:
```
[root@localhost _output]# ./nerdctl images
REPOSITORY                   TAG                                                                 IMAGE ID        CREATED              PLATFORM       SIZE         BLOB SIZE
calico/kube-controllers      v3.19.1                                                             904458fe1bd5    About an hour ago    linux/amd64    57.8 MiB     23.9 MiB
calico/node                  v3.19.1                                                             bc4a631d553b    24 minutes ago       linux/amd64    165.3 MiB    56.0 MiB
calico/pod2daemon-flexvol    v3.19.1                                                             bcac7dc4f130    24 minutes ago       linux/amd64    20.8 MiB     8.9 MiB
k8s.gcr.io/pause             3.5                                                                 1ff6c18fbef2    About an hour ago    linux/amd64    668.0 KiB    294.4 KiB
sha256                       5660150975fb82f006d013fdb583b628d3d340ff64775ca3111ca28f523e32df    bcac7dc4f130    24 minutes ago       linux/amd64    20.8 MiB     8.9 MiB
sha256                       5d3d5ddc8605ded8f69d76ee488072c7d02c32a8e4e8b34640a884c6eb939c0a    904458fe1bd5    About an hour ago    linux/amd64    57.8 MiB     23.9 MiB
sha256                       c4d75af7e098eef5142c65b8fab5b3ff0e5c614ce72e91b8c44ba20249b596ef    bc4a631d553b    24 minutes ago       linux/amd64    165.3 MiB    56.0 MiB
sha256                       ed210e3e4a5bae1237f1bb44d72a05a2f1e5c6bfe7a7e73da179e2534269c459    1ff6c18fbef2    About an hour ago    linux/amd64    668.0 KiB    294.4 KiB

```
if use nerdctl save bc 1f -o test.tar will not work (four ID start with bc  )

Reply:
Keep same with docker.

Reply:
`nerdctl load` support multi image import so we should support multi-image save